client/state/login/actions.js

```diff
@@ -109,6 +112,23 @@ export const remoteLoginUser = loginLinks => {
 	);
 };
 
+async function postLoginRequest( action, bodyObj ) {
+	const response = await fetch(
+		localizeUrl( `https://wordpress.com/wp-login.php?action=${ action }` ),
+		{
+			method: 'POST',
+			credentials: 'include',
+			headers: { Accept: 'application/json', 'Content-Type': 'application/x-www-form-urlencoded' },
+			body: stringify( bodyObj ),
+		}
+	);
+
+	if ( response.ok ) {
+		return { body: await response.json() };
+	}
+	throw new HTTPError( response, await response.body );
```

Comment:
`response.body` is not a `Promise`, it's a `ReadableStream` and doesn't seem to be thenable.

Only `response.json()` returns a promise: apparently it reads the response stream (async) and then parses it as JSON (sync).

If we want to be correct, we need to construct the error asynchronously:
```js
throw await HTTPError.create( response );

class HTTPError {
  static create( response ) {
    return response.json()
      .catch( () => null )
      .then( bodyJSON => new HTTPError( response, bodyJSON ) )
  }
}
```

Reply:
Sorry, I meant to use `response.text()` here. That one is a thenable and should work fine the way I was using it.

Reply:
Validated that this is working by making a request with the wrong password. The response correctly gets parsed and displayed as a specific error message, rather than a generic error message:

![image](https://user-images.githubusercontent.com/409615/62284188-120e8880-b44b-11e9-9592-44c63d465318.png)